tests/cases/compiler/inKeywordTypeguard.ts

```diff
@@ -273,10 +273,69 @@ function f9(x: object) {
     }
 }
 
+function f10(x: { a: unknown }) {
+    if ("a" in x) {
+        x;
+    }
+    else {
+        x;

```

Comment:
I don't quite understand why this would **not** be `never` here. Those calls are invalid:
```ts
declare const o4: { a?: number }
f10(o4) // error
declare const o5: {}
f10(o5) // error
```
So how, type-wise, could this be reached? After all, `in` should check the presence of a property and not that it can possibly be `undefined` 🤔

Reply:
This was an attempt to make feature detection code easier to write, but it seems to just trade one set of problems for another. See my comment [here](https://github.com/microsoft/TypeScript/issues/50954#issuecomment-1258856194). I'm rolling the change back and just fixing the issue with `SymbolFlags.Optional` computation for intersections.